qemu/tests/trim_support_test.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+import re
+
+from avocado.utils import process
+from virttest import utils_test
+from virttest import error_context
+from virttest import data_dir
+from virttest import utils_disk
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test disk trimming in windows guest
+    1) boot the vm with a data disk
+    2) format the data disk without quick mode
+    3) check the disk file size in host, and record for compare
+    4) trim the data disk in guest
+    5) check the disk file again in host, the file size should shrink
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def _get_size_value(size_str):
+        """
+        Get size value form size string
+
+        :param size_str: data size string
+        :return: the size numeric value, measured by MB
+        """
+        size = int(re.match(r'^\d+', size_str).group())
+        if "G" in size_str:
+            size *= 1024
+        return size
+
+    def _disk_size_smaller(ori_size):
+        """
+        Check till the disk size becomes smaller than ori_size
+        :param ori_size: original size to compare to
+        :return: new size if it smaller than ori_size, else None
+        """
+        output = process.system_output(host_check_cmd, shell=True).decode()
```

Comment:
you can call  qemu-img --output=json to read image size directly.

Reply:
It will "write lock" when the guest is on, so cannot check with qemu-img

Reply:
-U option can ignore that lock.